versions/development/SPEC.md

```diff
@@ -2646,6 +2648,30 @@ In JSON, the inputs to the workflow in the previous section might be:
 
 It's important to note that the type in JSON must be coercible to the WDL type.  For example `wf.int_val` expects an integer, but if we specified it in JSON as `"wf.int_val": "three"`, this coercion from string to integer is not valid and would result in a coercion error.  See the section on [Type Coercion](#type-coercion) for more details.
 
+## Specifying / Overriding Runtime Attributes in JSON
+
+Workflow runtime attributes may additionally be specified as key/value pairs within a JSON input file. It's recommended that this should be a separate inputs file called `runtime.json`. The mapping from JSON or YAML values to WDL values is codified similar to the serialization of task inputs section, however with an additional runtime tag to avoid name collisions and allow for unspecified attributes to be set by the runtime input file. Runtime attributes do not need to be specified in the task defintion to be overidden or set by the JSON.
+
+In JSON, the user may be able to specify task-specific runtime attributes similar to the following:
+
+```json
+{
+  "wf.t1.runtime.memory": "16 GB",
+  "wf.t2.runtime.cpus": 2,
+  "wf.t2.runtime.disks": "100 GB",
+  "wf.t3.runtime.arbitrary_key": ["arbitrary", "value"]
+}
+```
+
+As the runtime section consists of key/value pairs, it is the user's responsibility to ensure they provide the correct coercible type for the backend they are targeting. Expressions are disallowed within the `runtime.json`, hence no input will be coerced to an expression. See the section on [Type Coercion](#type-coercion) for more details.
```

Comment:
It's not that expressions are not *allowed*, it's that they're not evaluated. For instance you can put in `1+1` as a runtime attribute value, it just won't end up as `2` downstream.

Reply:
Yeah I see how my initial wording is a little ambiguous, I've modified it (in 26537d5c5058e8278dd157f52759cc5f2c5d135c) to read:

> An input value should not be coerced to an expression, hence values provided within the `runtime.json` should be evaluated.

I think it is clearer to read that what you provide _shouldn't_ be evaluated (like if you provided `"my test ${1+1} expression"`, it would remain), and the language better aligns with the rest of the spec.

Reply:
Yes, I think you should say they shouldn't be evaluated. But right now your wording says "values provided within the runtime.json should be evaluated", which is incorrect.

Reply:
... maybe I shouldn't add attention to detail on my resume anymore (fixed).